src/cogent3/app/data_store_new.py

```diff
@@ -272,53 +287,150 @@ def completed(self) -> list[DataMember]:
 
     @property
     def not_completed(self) -> list[DataMember]:
-        ic_dir = self.source / _NOT_COMPLETED_TABLE
+        nc_dir = self.source / _NOT_COMPLETED_TABLE
         return (
             [
                 DataMember(self, Path(_NOT_COMPLETED_TABLE) / m.name)
-                for m in ic_dir.glob("*.json")
+                for m in nc_dir.glob("*.json")
             ]
-            if ic_dir.exists()
+            if nc_dir.exists()
             else []
         )
 
     def write(self, unique_id: str, data: str) -> None:
         super().write(unique_id, data)
+        # check suffix compatible with this datastore
```

Comment:
there is a lot of duplication between these write methods. I suggest we create a `_write()` method that generalises this logic, taking the expected suffix for each output type etc.. For example
```python
def _write(self, subdir: str, identifier: str, suffix: str, data: str, mdf5: bool) -> None:
    sfx, cmp = get_format_suffixes(unique_id)
    if sfx != suffix:
        ...
    with open_(self.source / subdir) as out:
        ...
    if md5:
        ...
```
Then get all the write methods to use it
```python
def write(self, unique_id: str, data: str) -> None:
        self._write("", unique_id, self.suffix, data, True)

def write_log(self, unique_id: str, data: str) -> None:
        self._write(_LOG_TABLE, unique_id, "log", data, False)

```